rootex/core/resource_files/custom_material_resource_file.h

```diff
@@ -53,4 +56,12 @@ class CustomMaterialResourceFile : public MaterialResourceFile
 	bool save() override;
 	void draw() override;
 	void drawTextureSlots(const char* label, Vector<Ref<ImageResourceFile>>& textures);
+
+	float getFloat(int index);
+	Vector<float> getFloat3(int index);
+	Vector<float> getColor(int index);
```

Comment:
A Vector is not a mathematical vector. Vector is just a std::vector. Mathematical vectors are Vector2, Vector3 and Vector4.